tools/flaky-test-reporter/main.go

```diff
@@ -0,0 +1,67 @@
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// flaky-test-reporter collects test results from continuous flows,
+// identifies flaky tests, tracking flaky tests related github issues, 
+// and sends slack notifications.
+
+package main
+
+import (
+	"flag"
+	"log"
+	"os"
+
+	"github.com/knative/test-infra/shared/prow"
+)
+
+func main() {
+	serviceAccount := flag.String("service-account", os.Getenv("GOOGLE_APPLICATION_CREDENTIALS"), "JSON key file for service account to use")
+	dryrun         := flag.Bool("dry-run", false, "dry run switch")
+	flag.Parse()
+
+	if nil != dryrun && true == *dryrun {
+		log.Printf("running in [dry run mode]")
+	}
+	
+	prow.Initialize(*serviceAccount) // Explicit authenticate with gcs Client
+
+	// Clean up local artifacts directory, this will be used later for artifacts uploads
+	err := os.RemoveAll(prow.GetLocalArtifactsDir()) // this function returns nil if path not found
+	if nil == err {
+		if _, err = os.Stat(prow.GetLocalArtifactsDir()); os.IsNotExist(err) {
+			err = os.MkdirAll(prow.GetLocalArtifactsDir(), 0777)
+		}
+	}
+	if nil != err {
+		log.Fatalf("failed preparing local artifacts directory '%s'", err.Error())
+	}
+
+	var repoDataAll []*RepoData
+	for _, jc := range jobConfigs {
+		log.Printf("collecting results for repo '%s'\n", jc.Repo)
+		rd, err := collectTestResultsForRepo(&jc)
+		if nil != err {
+			log.Printf("%s", err.Error())
```

Comment:
Should this be fatal? I'm afraid that if you just print it, it will be overlooked.

```suggestion
			log.Printf("Error collecting results for repo: %v", err)
```

Reply:
Great suggestion! I have updated to use Fatalf everywhere in data collection phase, nothing should be done if the data is inaccurate